joint_trajectory_controller/test/joint_trajectory_controller_test.cpp

```diff
@@ -414,117 +558,104 @@ TEST_F(JointTrajectoryControllerTest, invalidMessages)
 
-  // Wait until done
-  ASSERT_TRUE(waitForState(action_client, SimpleClientGoalState::SUCCEEDED, long_timeout));
-  EXPECT_EQ(action_client->getResult()->error_code, control_msgs::FollowJointTrajectoryResult::SUCCESSFUL);
-  ros::Duration(0.5).sleep(); // Allows values to settle to within EPS, especially accelerations
+  auto timeout{getTrajectoryDuration(traj_goal.trajectory) + ros::Duration(TIMEOUT_ACTION_RESULT_S)};
+  ASSERT_TRUE(waitForActionResult(action_client, timeout));
+  EXPECT_TRUE(checkActionGoalState(action_client, SimpleClientGoalState::SUCCEEDED));
+  EXPECT_TRUE(checkActionResultErrorCode(action_client, control_msgs::FollowJointTrajectoryResult::SUCCESSFUL));
+
+  EXPECT_TRUE(waitForStop()); // Allows values to settle within JOINT_STATES_COMPARISON_EPS, especially accelerations
 
   // Validate state topic values
   StateConstPtr state = getState();
-  for (unsigned int i = 0; i < n_joints; ++i)
-  {
-    EXPECT_NEAR(traj.points.back().positions[i],     state->desired.positions[i],     EPS);
-    EXPECT_NEAR(traj.points.back().velocities[i],    state->desired.velocities[i],    EPS);
-    EXPECT_NEAR(traj.points.back().accelerations[i], state->desired.accelerations[i], EPS);
 
-    EXPECT_NEAR(traj.points.back().positions[i],  state->actual.positions[i],  EPS);
-    EXPECT_NEAR(traj.points.back().velocities[i], state->actual.velocities[i], EPS);
+  EXPECT_TRUE(trajectoryPointsAlmostEqual(traj.points.back(), state->desired));
+  EXPECT_TRUE(vectorsAlmostEqual(traj.points.back().positions, state->actual.positions));
+  EXPECT_TRUE(vectorsAlmostEqual(traj.points.back().velocities, state->actual.velocities));
 
-    EXPECT_NEAR(0.0, state->error.positions[i],  EPS);
-    EXPECT_NEAR(0.0, state->error.velocities[i], EPS);
-  }
+  std::vector<double> zero_vec(n_joints, 0.0);
+  EXPECT_TRUE(vectorsAlmostEqual(state->error.positions, zero_vec));
+  EXPECT_TRUE(vectorsAlmostEqual(state->error.velocities, zero_vec));
 }
 
 // Joint reordering ////////////////////////////////////////////////////////////////////////////////////////////////////
 
 TEST_F(JointTrajectoryControllerTest, jointReordering)
 {
-  ASSERT_TRUE(initState());
-  ASSERT_TRUE(action_client->waitForServer(long_timeout));
-
   // Message joints are ordered differently than in controller
   ActionGoal reorder_goal = traj_home_goal;
   std::swap(reorder_goal.trajectory.joint_names[0], reorder_goal.trajectory.joint_names[1]);
   reorder_goal.trajectory.points.front().positions[0] = M_PI / 4.0;
   reorder_goal.trajectory.points.front().positions[1] = 0.0;
 
-
   reorder_goal.trajectory.header.stamp = ros::Time(0); // Start immediately
   action_client->sendGoal(reorder_goal);
-  ASSERT_TRUE(waitForState(action_client, SimpleClientGoalState::SUCCEEDED, long_timeout));
-  ros::Duration(0.5).sleep(); // Allows values to settle to within EPS, especially accelerations
+
+  auto timeout{getTrajectoryDuration(reorder_goal.trajectory) + ros::Duration(TIMEOUT_ACTION_RESULT_S)};
```

Comment:
I think you may be overusing {} a little bit :) 
```suggestion
  const auto timeout = getTrajectoryDuration(reorder_goal.trajectory) + ros::Duration(TIMEOUT_ACTION_RESULT_S);
```
is just so much more readable and there's not loss, `{}` is generally recommended for member variables, while this is local *and* in a test so I wouldn't push to beautify

Reply:
Done.